terraform/services/backup-plan/primary.tf

```diff
@@ -0,0 +1,211 @@
+          "kms:CreateGrant",
+          "kms:ListGrants",
+          "kms:RevokeGrant"
+        ],
+        "Resource" : "*",
+        "Condition" : {
+          "Bool" : {
+            "kms:GrantIsForAWSResource" : "true"
+          }
+        }
+      }
+    ]
+    Version = "2012-10-17"
+  })
+}
+
+data "aws_kms_alias" "primary_kms_alias" {
+  for_each = toset(local.apps)
+  name     = lower("alias/${each.value}-${var.env}")
+}
+
+data "aws_kms_key" "primary_kms_key" {
+  for_each = toset(local.apps)
+  key_id   = data.aws_kms_alias.primary_kms_alias[each.value].target_key_id
+}
+
+resource "aws_backup_vault" "primary_backup_vault" {
+  for_each    = toset(local.apps)
+  name        = "${var.vault_name}_${each.value}"
+  kms_key_arn = data.aws_kms_key.primary_kms_key[each.value].arn
+}
+
+data "aws_iam_policy_document" "primary_backup_policy" {
+  for_each = toset(local.apps)
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_caller_identity.current.account_id]
+    }
+
+    actions = [
+      "backup:CopyIntoBackupVault",
+    ]
+
+    resources = [aws_backup_vault.primary_backup_vault[each.value].arn]
+  }
+}
+
+resource "aws_backup_vault_policy" "primary_backup_vault_policy" {
+  for_each          = toset(local.apps)
+  backup_vault_name = aws_backup_vault.primary_backup_vault[each.value].name
+  policy            = data.aws_iam_policy_document.primary_backup_policy[each.value].json
+}
+
+resource "aws_backup_plan" "aws_backup_plan" {
+  for_each = toset(local.apps)
+  name     = "cdap_managed_backup_plan_${each.value}"
+  #only the 4hr rule should be copied to secondary
+  rule {
+    rule_name         = "4Hourly_1"
+    target_vault_name = aws_backup_vault.primary_backup_vault[each.value].name
+    schedule          = "cron(0 */4 * * ? *)"
+    start_window      = 60
+    completion_window = 180
+    copy_action {
+      destination_vault_arn = aws_backup_vault.secondary_backup_vault[each.value].arn
+    }
+
+    lifecycle {
+      delete_after = 1
+    }
+  }
+
+  rule {
+    rule_name         = "Daily_7"
+    target_vault_name = aws_backup_vault.primary_backup_vault[each.value].name
+    schedule          = "cron(0 4 * * ? *)"
+    start_window      = 60
+    completion_window = 180
+
+    lifecycle {
+      delete_after = 7
+    }
+  }
+
+  rule {
+    rule_name         = "Weekly_35"
+    target_vault_name = aws_backup_vault.primary_backup_vault[each.value].name
+    schedule          = "cron(0 0 ? * SAT *)"
+    start_window      = 60
+    completion_window = 180
+
+    lifecycle {
+      delete_after = 35
+    }
+  }
+
+  rule {
+    rule_name         = "Monthly_90"
+    target_vault_name = aws_backup_vault.primary_backup_vault[each.value].name
+    schedule          = "cron(0 0 1 * ? *)"
+    start_window      = 60
+    completion_window = 180
+
+    lifecycle {
+      delete_after = 90
+    }
+  }
+
+  depends_on = [
+    aws_backup_vault.secondary_backup_vault
+  ]
```

Comment:
This depends_on should not be necessary since the resource is referenced in the copy_action block above.